diff -u exim/src/Local/Makefile exim-patch/src/Local/Makefile
--- exim/src/Local/Makefile	2022-11-17 12:41:05
+++ exim-patch/src/Local/Makefile	2022-11-17 12:44:50
@@ -132,7 +132,7 @@
 # deliveries. (Local deliveries run as various non-root users, typically as the
 # owner of a local mailbox.) Specifying these values as root is not supported.
 
-EXIM_USER=
+EXIM_USER=ubuntu
 
 # If you specify EXIM_USER as a name, this is looked up at build time, and the
 # uid number is built into the binary. However, you can specify that this