--- Makefile	2022-11-17 14:43:00
+++ Makefile-patch	2022-11-17 14:43:14
@@ -132,7 +132,7 @@
 # deliveries. (Local deliveries run as various non-root users, typically as the
 # owner of a local mailbox.) Specifying these values as root is not supported.
 
-EXIM_USER=
+EXIM_USER=ubuntu
 
 # If you specify EXIM_USER as a name, this is looked up at build time, and the
 # uid number is built into the binary. However, you can specify that this
@@ -203,7 +203,7 @@
 
 # Uncomment the following lines if you want
 # to build Exim without any TLS support (either OpenSSL or GnuTLS):
-# DISABLE_TLS=yes
+DISABLE_TLS=yes
 # Unless you do this, you must define one of USE_OPENSSL or USE_GNUTLS
 # below.
 
@@ -277,7 +277,7 @@
 
 
 # Uncomment the following line to remove support for TLS Resumption
-# DISABLE_TLS_RESUME=yes
+DISABLE_TLS_RESUME=yes
 
 
 ###############################################################################

