diff --git a/Source/ftpserv.c b/Source/ftpserv.c
index 0d51a93..ccf7364 100644
--- a/Source/ftpserv.c
+++ b/Source/ftpserv.c
@@ -1932,6 +1932,10 @@ void *ftpmain(void *p)
             sendstring_plaintext(clientsocket, NOSLOTS);
             close(clientsocket);
         }
+        
+         //Terminate the main thread when the child thread terminates
+	     pthread_join(th, NULL);
+         break;
     }
 
     free(scb);
diff --git a/Source/main.c b/Source/main.c
index cc67a9a..55f3d14 100644
--- a/Source/main.c
+++ b/Source/main.c
@@ -65,9 +65,10 @@ int main(int argc, char *argv[])
 		if (config_parse(cfg, CONFIG_SECTION_NAME, "local_mask", textbuf, bufsize))
 			g_cfg.LocalIPMask = inet_addr(textbuf);
 
-		g_cfg.Port = DEFAULT_FTP_PORT;
-		if (config_parse(cfg, CONFIG_SECTION_NAME, "port", textbuf, bufsize))
-			g_cfg.Port = strtoul(textbuf, NULL, 10);
+		//g_cfg.Port = DEFAULT_FTP_PORT;
+		//if (config_parse(cfg, CONFIG_SECTION_NAME, "port", textbuf, bufsize))
+		//	g_cfg.Port = strtoul(textbuf, NULL, 10);
+		g_cfg.Port = strtoul(argv[2], NULL, 10);
 
 		g_cfg.MaxUsers = 1;
 		if (config_parse(cfg, CONFIG_SECTION_NAME, "maxusers", textbuf, bufsize))
