diff --git a/apps/openssl.c b/apps/openssl.c
index a3056c799f..bc15d31a58 100644
--- a/apps/openssl.c
+++ b/apps/openssl.c
@@ -29,6 +29,8 @@
 #include "apps.h"
 #include "progs.h"
 
+#include "../fuzz/rand.inc"
+
 /*
  * The LHASH callbacks ("hash" & "cmp") have been replaced by functions with
  * the base prototypes (we cast each variable inside the function to the
@@ -246,6 +248,8 @@ int main(int argc, char *argv[])
     arg.argv = NULL;
     arg.size = 0;
 
+    FuzzerSetRand();
+
     /* Set up some of the environment. */
     bio_in = dup_bio_in(FORMAT_TEXT);
     bio_out = dup_bio_out(FORMAT_TEXT);
