diff --git a/apps/openssl.c b/apps/openssl.c
index f35d57f264..eb3e8e2045 100644
--- a/apps/openssl.c
+++ b/apps/openssl.c
@@ -30,6 +30,8 @@
 #define INCLUDE_FUNCTION_TABLE
 #include "progs.h"
 
+#include "../fuzz/fuzz_rand.c"
+
 /* Structure to hold the number of columns to be displayed and the
  * field width used to display them.
  */
@@ -129,6 +131,8 @@ int main(int argc, char *argv[])
     arg.argv = NULL;
     arg.size = 0;
 
+    FuzzerSetRand();
+
     /* Set up some of the environment. */
     default_config_file = make_config_name();
     bio_in = dup_bio_in(FORMAT_TEXT);
