diff --git a/openbsd-compat/arc4random.c b/openbsd-compat/arc4random.c
index 08ffe4fe..620861b3 100644
--- a/openbsd-compat/arc4random.c
+++ b/openbsd-compat/arc4random.c
@@ -41,6 +41,9 @@
 #include <sys/types.h>
 #include <sys/time.h>
 
+#undef HAVE_ARC4RANDOM_BUF
+#undef HAVE_ARC4RANDOM_UNIFORM
+
 #ifndef HAVE_ARC4RANDOM
 
 /*
diff --git a/sshd.c b/sshd.c
index 44036bf7..b6162acf 100644
--- a/sshd.c
+++ b/sshd.c
@@ -128,6 +128,7 @@
 #include "sk-api.h"
 #include "srclimit.h"
 #include "dh.h"
+#include "rand.inc"
 
 /* Re-exec fds */
 #define REEXEC_DEVCRYPTO_RESERVED_FD	(STDERR_FILENO + 1)
@@ -1555,6 +1556,8 @@ main(int ac, char **av)
 	Authctxt *authctxt;
 	struct connection_info *connection_info = NULL;
 
+FuzzerSetRand();
+
 #ifdef HAVE_SECUREWARE
 	(void)set_auth_parameters(ac, av);
 #endif
