diff --git a/dcmqrdb/apps/dcmqrscp.cc b/dcmqrdb/apps/dcmqrscp.cc
index 8b29ed13e..2b9cb4aed 100644
--- a/dcmqrdb/apps/dcmqrscp.cc
+++ b/dcmqrdb/apps/dcmqrscp.cc
@@ -106,8 +106,7 @@ static void mangleAssociationProfileKey(OFString& key)
 int
 main(int argc, char *argv[])
 {
-  OFString opt_configFileName = OFStandard::getDefaultConfigurationDir();
-  opt_configFileName += "dcmqrscp.cfg";
+  OFString opt_configFileName = "./dcmqrscp.cfg";
   OFCondition cond = EC_Normal;
   OFCmdUnsignedInt overridePort = 0;
   OFCmdUnsignedInt overrideMaxPDU = 0;
